projects/Magnetosphere/Magnetosphere.cpp

```diff
@@ -464,6 +464,9 @@ namespace projects {
                      if(technicalGrid.get(x, y, z)->sysBoundaryFlag == sysboundarytype::SET_MAXWELLIAN ) {
                         for (int i = 0; i < fsgrids::bgbfield::N_BGB; ++i) {
                            BgBGrid.get(x,y,z)->at(i) = 0;
+			   if ( (this->dipoleType==4) && (P::isRestart == false) ) {
+			      perBGrid.get(x,y,z)->at(i) = 0;
+			   }
```

Comment:
No, this will break spectacularly. You are accessing perBGrid with indices over fsgrid::bgbfield. Unless they have the same size?

Reply:
This will go haywire, perBGrid has 3 entries, bgBGrid has 17 or so.

Reply:
Ah whoops, too coding fingers were too quick for my brain